digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/lane/LaneService.scala

```diff
@@ -0,0 +1,609 @@
+      }
+
+      if (changeSet.adjustedVVHChanges.nonEmpty)
+        logger.info("Saving adjustments for lane/link ids=" + changeSet.adjustedVVHChanges.map(a => "" + a.laneId + "/" + a.linkId).mkString(", "))
+
+      changeSet.adjustedVVHChanges.foreach { adjustment =>
+        dao.updateMValuesChangeInfo(adjustment.laneId, (adjustment.startMeasure, adjustment.endMeasure), adjustment.vvhTimestamp, VvhGenerated)
+      }
+
+      val ids = changeSet.expiredLaneIds.toSeq
+      if (ids.nonEmpty)
+        logger.info("Expiring ids " + ids.mkString(", "))
+      ids.foreach(id => dao.updateExpiration(id, VvhGenerated))
+
+      if (changeSet.adjustedSideCodes.nonEmpty)
+        logger.info("Saving SideCode adjustments for lane/link ids=" + changeSet.adjustedSideCodes.map(a => "" + a.laneId).mkString(", "))
+
+      changeSet.adjustedSideCodes.foreach { adjustment =>
+        adjustedSideCode(adjustment)
+      }
+
+      if (changeSet.valueAdjustments.nonEmpty)
+        logger.info("Saving value adjustments for assets: " + changeSet.valueAdjustments.map(a => "" + a.lane.id).mkString(", "))
+      changeSet.valueAdjustments.foreach { adjustment =>
+        updatePersistedLaneAttributes( adjustment.lane.id, adjustment.lane.attributes, adjustment.lane.modifiedBy.get)
+      }
+    }
+  }
+
+  def adjustedSideCode(adjustment: SideCodeAdjustment): Unit = {
+    val oldAsset = getPersistedLanesByIds( Set(adjustment.laneId), newTransaction = false).head
+    val roadLink = roadLinkService.getRoadLinkAndComplementaryFromVVH(oldAsset.linkId, newTransaction = false).getOrElse(throw new IllegalStateException("Road link no longer available"))
+
+    val newLane = PersistedLane (0, oldAsset.linkId, adjustment.sideCode.value, oldAsset.laneCode, roadLink.municipalityCode,
+      oldAsset.startMeasure, oldAsset.endMeasure, Some(VvhGenerated), Some( DateTime.now() ),
+      None,None, expired = false, vvhClient.roadLinkData.createVVHTimeStamp(), oldAsset.geomModifiedDate, oldAsset.attributes)
+
+    dao.updateExpiration(oldAsset.id, VvhGenerated)
+   createWithoutTransaction( newLane, VvhGenerated )
+  }
+
+  def updatePersistedLaneAttributes( id: Long, attributes: LanePropertiesValues, username: String) = {
+    attributes.properties.map { prop =>
+      dao.updateLaneAttributes(id, prop, username)
+    }
+  }
+
+  def deleteMultipleLanes(ids: Set[Long]) = {
+    withDynTransaction {
+      ids.map(id => dao.deleteEntryLane(id))
+    }
+  }
+
+  def deleteEntryLane(id: Long ): Unit = {
+    withDynTransaction {
+      dao.deleteEntryLane(id)
+    }
+  }
+
+  def multipleLanesToHistory (ids: Set[Long], username: String):  Set[Long] ={
+    withDynTransaction {
+      ids.map(id => dao.updateLaneExpiration(id, username) )
+    }
+    ids
+  }
+
+  def laneToHistory (id: Long, username: String): Long ={
+    withDynTransaction {
+      dao.updateLaneExpiration(id, username)
+    }
+    id
+  }
```

Comment:
laneToHistory is not being used, delete please

Reply:
I think it will be helpful in a near future

Reply:
To be deleted, please

Reply:
Done